src/kernels/MIOpenBatchNormBwdSpatial.cl

```diff
@@ -815,8 +815,8 @@ MIOpenBatchNormBwdSpatialDScaleDBias(const __global _FLOAT* x_in,
     {
         unsigned int betaindex  = cidx + ygrp_sz * ygrp_id + 6;
         unsigned int gammaindex = cidx + ygrp_sz * ygrp_id + 4;
-        buff[gammaindex]        = (_FLOAT_PREC)dscale;
-        buff[betaindex]         = (_FLOAT_PREC)dbias;
+        buff[gammaindex]        = FLOAT2FLOATPREC(dscale);
+        buff[betaindex]         = FLOAT2FLOATPREC(dbias);
```

Comment:
this is odd since buff is _FLOAT*.  but then a bunch of the functions around here don't seem right  like the ones that only use _FLOAT so they better never be used for bf16.

Reply:
Yea these look incorrect for BF16, I think we need to change this.